File: Front_End/src/app/mg/modules/pesage.ts
// MS Emballage
import { File } from "./file";
import { Recu } from "./recu";

export class Pesage {

  id: number;
  dateDeclaration: Date;
  agentBallance : number;

  vehiculeMatricule: string;
  conducteurCin: string;
  vendeurCin: string;

  produit: number;
  typeProduit: number;
  sousTypeProduit: number;

  hangar: number;
  provenance: string;

  emballage: number;
  nombreUnite: number;
  poidsEmballage: number;

  poidsTotal: number;
  poidsNet: number;

  poidVideVehicule: number;

  prixReference: number;
  montantTotal: number;
  taxe: number;
  taxePayer: number;

  files: File[];
  recu : Recu;

}

import { AgentBalance } from "./AgentBalance";
import { Vehicule } from "./vehicule";

import { Conducteur } from "./conducteur";
import { Vendeur } from "./Vendeur";

import { Produit } from "./Produit";
import { TypeProduit } from "./TypeProduit";
import { SousTypeProduit } from "./SousTypeProduit";

import { Hangarbase } from "./Hangar";
import { Emballage } from "./emballage";


export class PesageLord {

  id: number;
  dateDeclaration: Date;
  agentBallance : AgentBalance;

  vehiculeMatricule: Vehicule;
  conducteurCin: Conducteur;
  vendeurCin: Vendeur;

  produit: Produit;
  typeProduit: TypeProduit;
  sousTypeProduit: SousTypeProduit;

  hangar: Hangarbase;
  provenance: string;

  emballage: Emballage;
  nombreUnite: number;
  poidsEmballage: number;

  poidsTotal: number;
  poidsNet: number;

  poidVideVehicule: number;

  prixReference: number;
  montantTotal: number;
  taxe: number;
  taxePayer: number;

  files: File[];
  recu : Recu;

}

File: Back_End/sic.mg.ballance/src/main/java/ma/sic/mg/ballance/entities/Pesage.java
package ma.sic.mg.ballance.entities;


import java.io.Serializable;
import java.util.Date;
import java.util.List;

import javax.persistence.*;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Pesage implements Serializable {

	private static final long serialVersionUID = 1L;

	@Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
	
	private String agentBallance;
    private Date dateDeclaration;

    private String vehiculeMatricule;
    private String conducteurCin;
    private String vendeurCin;
    
    private long typeProduit;
    private long sousTypeProduit;

    private String provenance;
    private int hangar;

    private long emballage;
    private double poidsEmballage;
    private int nombreUnite;
        
    private double poidsTotal;
    private double poidsNet;
    
    private double poidVideVehicule;
    
    private double prixReference;
    private double montantTotal;
    private double taxe;
    private double taxePayer;
    
    
    @OneToMany(mappedBy = "pesage", cascade = CascadeType.ALL)
    private List<FilePjPesage> files;
    
    @OneToOne
	private Recu recu;
	
}
